reactive/containerd.py

```diff
@@ -248,6 +269,50 @@ def merge_custom_registries(config_directory, custom_registries,
     return registries
 
 
+def invalid_custom_registries(custom_registries):
+    """
+    Validate custom registries from config.
+
+    :param str custom_registries: juju config for custom registries
+    :return: error string for blocked status if condition exists, None otherwise
+    :rtype: Optional[str]
+    """
+    try:
+        registries = _registries_list(custom_registries)
+    except json.JSONDecodeError:
+        return "Failed to decode json string"
```

Comment:
This will squash the JSONDecodeError without logging it. That could omit details that are useful to our users and developers alike. For example:

```
>>> json.loads('[{"url": "www.docker.io}]')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 10 (char 9)
```

I recommend logging the exception with `log(traceback.format_exc())`.